UI/Buttons/FavoriteButton.swift

```diff
@@ -28,6 +28,7 @@ struct FavoriteButton: View {
         let symbol: SFSymbol = isFavorite ? .heartFill : .heart
         let text = isFavorite ? textUnfavorite : textFavorite
         AsyncButton {
+            Haptics.impactOccurred()
```

Comment:
This does not work on my phone and I see no errors in xcode. Not sure what's wrong.

Reply:
Works on mine. I'd keep it and you can try debugging it later.

Reply:
It seems that from ios17 the new way is to use `.sensoryFeedback()` instead, so I'd try that? 

https://developer.apple.com/documentation/swiftui/sensoryfeedback/
https://www.hackingwithswift.com/books/ios-swiftui/adding-haptic-effects

Reply:
Just tested it again and it works. But it's so weak I actually have to focus just to feel it. Using the view method with `.success` would be much better imo.

I checked Music and it also uses `.success`. Although it's a bit delayed and waits until the star animation finishes.

Reply:
Yeah, Music has a fancy animation that we don't, so I opted for immediate haptic feedback on button tap.

I've changed it to have success on favoriting and just impact on unfavoriting.